index.rst

```diff
@@ -77,19 +85,39 @@ categorized as:
 Tag Fragments
 -------------
 
-In general, tags can be postfixed with an optional Cycle number or a
-``rest`` set of one-or-more underscore-separated fields.  The Cycle is
-currently unique to T&S RSP instances, and specifies the version of XML
-used in this image.  It will have the form ``c<digits>.<digits>`` where
-the first group of digits is the cycle number and the second group of
-digits is the build iteration for that cycle, e.g. ``c0019.001``.
+Primary
+^^^^^^^
+The primary version of a tag is derived from (and in most cases identical to) the EUPS tag of the underlying DM stack included in the image.
+It is always the first component of the tag.
+
+RSP Build Counter
+^^^^^^^^^^^^^^^^^
+
+Tags may include an identifier specifying an RSP image build count. 
+In general, any run of the GitHub Action that builds a sciplat-lab image will increment this counter.
+It is currently in the 2000s, and we do not anticipate it exceeding 10,000 during the lifetime of the project.
+We anticipate only including this counter in Release, Release Candidate, and Experimental builds derived from one of the former two categories, although it will continue to increment with each build regardless of type.
+
+If specified, this goes directly after the primary component.
+It takes the form ``rspX`` where ``X`` is an integer and is separated from the primary tag with an underscore.
+
+Tags can additionally be postfixed with an optional Cycle number or a ``rest`` set of one-or-more underscore-separated fields.
+
+Cycle
+^^^^^
+
+The Cycle is currently unique to T&S RSP instances, and specifies the version of XML used in this image.
+It will have the form ``c<digits>.<digits>`` where the first group of digits is the cycle number and the second group of digits is the build iteration for that cycle, e.g. ``c0019.001``.
+
+Rest
+^^^^
 
 The ``rest`` fragment will most often encode a build datestamp, but can
 be anything (but see the discussion in "Semantic Versions").
 
 For image types for which semantic versions are supported, both of these
-fields will end up packed into the ``build`` semver field.  If both are
-present in a single tag, the cycle must precede ``rest``.
+fields will end up packed into the ``build`` semver field.
```

Comment:
I think this needs more detail, since the semver field is an internal implementation detail of the controller and it's not at all obvious what this means. I think what you're trying to say is that versions will be sorted first by the cycle and then by `rest`, in purely alphanumeric order (not semantic version or integer order).

Reply:
Actually this bit doesn't have anything to do with the sorting.  This is: if you have a tag from which a "semantic version" (modulo not guaranteeing anything about inter-version compatibility) can be generated, both the "cycle" and "rest" data ends up packed into the semver "build" field.  It is an internal implementation detail so maybe we should just leave it out?